### PR TITLE
fix: zero volume errors

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
The general idea here is to let the player communicate with React via `setState` rather than to directly query the player (if possible).

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -228,12 +230,15 @@ export default class App extends React.Component {
...
-    });
+    this.setState(
+      {
+        audioConfig
+      },
+      () => {
+        // Save user volume to local storage
+        store.set(CODERADIO_VOLUME, maxVolume);
+      }
+    );
```
Comment (ojeytonwilliams) [inline src/components/App.js:241]: This hasn't changed, I just applied prettier.

File: src/components/App.js [hunk 2]
```diff
@@ -175,32 +176,33 @@ export default class App extends React.Component {
...
-        pullMeta: true
+        this.fadeUp();
       });
-
-      this.fadeUp();
     }
   }
 
   pause() {
-    if (this._player.paused) return;
```
Comment (ojeytonwilliams) [inline src/components/App.js:193]: `state.playing` should track this properly, so we should never need to check it directly.

**Review by huyenltnguyen — APPROVED** [review]

File: src/components/App.js [hunk 3]
```diff
@@ -272,7 +278,8 @@ export default class App extends React.Component {
     // If the volume is correctly set to the target, no need to change it
     if (currentVolume === this.state.audioConfig.targetVolume) {
       // If the audio is set to 0 and it’s been met, pause the audio
-      if (this.state.audioConfig.targetVolume === 0) this.pause();
+      if (this.state.audioConfig.targetVolume === 0 && this.state.pausing)
```
Comment (ahmadabdolsaheb) [inline src/components/App.js:281]: So when would be the time that the vol is zero and we would not want to pause?

Comment (ojeytonwilliams) [reply src/components/App.js:281]: If you're fading up then you shouldn't pause, even if the target volume is 0.  The idea is that most internet radios that I could find allowed you to toggle the play/pause state independently of the volume, and I thought we should provide a familiar UX.

**Pull request merged**